spec/instant.html

```diff
@@ -647,5 +623,40 @@ <h1>TemporalInstantToString ( _instant_, _timeZone_, _precision_ )</h1>
         1. Return the string-concatenation of _dateTimeString_ and _timeZoneString_.
       </emu-alg>
     </emu-clause>
+    <emu-clause id="sec-temporal-differencetemporalinstant" type="abstract operation">
+      <h1>
+        DifferenceTemporalInstant (
+          _operation_: ~since~ or ~until~,
+          _instant_: a Temporal.Instant,
+          _other_: an ECMAScript language value,
+          _options_: an ECMAScript language value,
+        ): either a normal completion containing a Temporal.Duration or an abrupt completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It computes the difference between the two times represented by _instant_ and _other_, optionally rounds it, and returns it as a Temporal.Duration object.</dd>
+      </dl>
+      <emu-alg>
+        1. Set _other_ to ? ToTemporalInstant(_other_).
+        1. If _operation_ is ~until~, then,
```

Comment:
For consistency, can we use the same logic for until vs. since handling that's used in all other AOs (especially `DifferenceTemporalPlainTime` whose behavior should be almost identical to this one except for overflow handling) ?  

Specifically, removing this swapping of `_other_` and `_instant_` and then using logic like we use elsewhere. Like this:

```
1. If _operation_ is ~since~, let _sign_ be -1. Otherwise, let _sign_ be 1.
```
then...
```
        1. If _operation_ is ~since~, then
          1. Set _roundingMode_ to ! NegateTemporalRoundingMode(_roundingMode_).
```
and finally: 

``` 
1. Let _result_ be ! BalanceDuration(0, 0, 0, 0, 0, 0, `sign` × _roundedNs_, _largestUnit_).
```

If there's a perf or other reason that this AO should use the swapping logic, then why doesn't  `DifferenceTemporalPlainTime` have the same optimization?

Reply:
What you are asking is not a refactoring of the current logic, but rather a different math. I am not opposing to do that but I lack of confidence to that kind of change. Just look at the current spec:
https://tc39.es/proposal-temporal/#sec-temporal.instant.prototype.since
https://tc39.es/proposal-temporal/#sec-temporal.instant.prototype.round

I simply copy the alg there and factor out. What you suggest is way more complicated than that and I do not feel I have the time to ensure such change is correct. 

I am also trubled by the "consistency" ideology.  "consistency" will need to end somewhere while the two different type are differeent. It is not avoidable some "consistency" will be broken to deal with type specific issue. WHER the "consistency" should end is the question and I feel we push too deep on this. I think "consistency" should be kept on public API level and someone want it to be in the AO function level. But inside alg of AO? I think that simply go too far. 

How about we first land this AO with if/swap . And if you or anyone feel comfortable of changing that logic w/o introduce error (which I do not feel confident) than you can do that later in a separate editorial PR.